examples/qml_minimal/CMakeLists.txt

```diff
@@ -33,68 +33,30 @@ endif()
 # ANCHOR: book_cmake_find_qmake
 # The path to the qmake executable path needs to be passed to the Rust
 # library's build script to ensure it uses the same installation of Qt as CMake.
-get_target_property(QMAKE Qt::qmake IMPORTED_LOCATION)
+# TODO: This has been removed, document it
 # ANCHOR_END: book_cmake_find_qmake
 
 # ANCHOR: book_cmake_find_corrosion
-find_package(Corrosion QUIET)
-if(NOT Corrosion_FOUND)
-    include(FetchContent)
-    FetchContent_Declare(
-        Corrosion
-        GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
-        GIT_TAG v0.5.0
-    )
-
-    FetchContent_MakeAvailable(Corrosion)
-endif()
+# TODO: Replace with fetch-content co cxx-qt-cmake
 # ANCHOR_END: book_cmake_find_corrosion
 
 # ANCHOR: book_cmake_use_corrosion
 set(CRATE qml_minimal)
-# Corrosion creates a CMake target with the same name as the crate.
-corrosion_import_crate(MANIFEST_PATH rust/Cargo.toml CRATES ${CRATE})
-
-# The Rust library's build script needs to be told where to output the
-# generated headers so CMake can find them. To do this, tell Corrosion
-# to set the CXXQT_EXPORT_DIR environment variable when calling `cargo build`.
-# Also, set the QMAKE environment variable to ensure the Rust library uses
-# the same installation of Qt as CMake.
-set(CXXQT_EXPORT_DIR "${CMAKE_CURRENT_BINARY_DIR}/cxxqt")
-corrosion_set_env_vars(${CRATE}
-    "CXXQT_EXPORT_DIR=${CXXQT_EXPORT_DIR}"
-    "QMAKE=${QMAKE}"
-    $<$<BOOL:${CMAKE_RUSTC_WRAPPER}>:RUSTC_WRAPPER=${CMAKE_RUSTC_WRAPPER}>
-)
-
-# Create an INTERFACE library target to link libraries to and add include paths.
-# Linking this to both the application and the tests avoids having to setup
-# the include paths and linked libraries for both of those.
-add_library(${APP_NAME}_lib INTERFACE)
-
-# Include the headers generated by the Rust library's build script. Each
-# crate gets its own subdirectory under CXXQT_EXPORT_DIR. This allows you
-# to include headers generated by multiple crates without risk of one crate
-# overwriting another's files.
-target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
-
-target_link_libraries(${APP_NAME}_lib INTERFACE
-    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
-    # otherwise the linker will discard the static variables that initialize it.
-    "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
-    Qt::Core
-    Qt::Gui
-    Qt::Qml
-    Qt::QuickControls2
-)
+# Corrosion (through CXX-Qt) creates a CMake target with the same name as the crate.
+cxxqt_import_crate(MANIFEST_PATH rust/Cargo.toml CRATES ${CRATE})
+target_link_libraries(${CRATE} INTERFACE Qt::Core Qt::Gui Qt::Qml Qt::QuickControls2)
+
+cxxqt_import_qml_module(${CRATE}_qml
```

Comment:
some of the examples you use `_qml` and some `qml` would be good to be consistent. Also don't know if we put part of the uri in the target name so that it's obvious this is related to the plugin ?

Reply:
Ah, right that was feedback I got from Milian.
Especially in the examples that we show in the book, we'll probably not want to use this added ${CRATE} variable, or at least use another separator like `_` when appending something to that, as it's just hard to read otherwise.
I'll add that everywhere.

Reply:
Done